File: recursividade/python/checar-ordenacao.py
def is_sorted(lst):
    if len(lst) <= 1:
        return True

    if lst[0] > lst[1]:
        return False

    tail = lst[1:]
    return is_sorted(tail)


print(is_sorted([]))
print(is_sorted([15.0, 20.0, 22.0, 31.0, 40.0]))
print(is_sorted([15.0, 20.0, 22.0, 21.0, 40.0]))

File: recursividade/java/src/ChecarOrdenacao.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class ChecarOrdenacao {
	
	public static boolean isSorted(List<Double> list) {
        if (list.size() <= 1) {
            return true;
        }

        if (list.get(0) > list.get(1)) {
            return false;
        }

        List<Double> tail = list.subList(1, list.size());
        return isSorted(tail);
    }

    public static void main(String[] args) {
        System.out.println(isSorted(new ArrayList<>()));
        System.out.println(isSorted(Arrays.asList(15.0, 20.0, 22.0, 31.0, 40.0)));
        System.out.println(isSorted(Arrays.asList(15.0, 20.0, 22.0, 21.0, 40.0)));
    }
}
